doc/source/programs/gdalwarp.rst

```diff
@@ -534,6 +534,17 @@ of an individual source pixel is determined by the product of the weights
 determined for its row and column; the diagonal Cartesian distance is not
 calculated.
 
+Multiple input files
+--------------------
+
+When multiple inputs are provided to :program:`gdalwarp`, they are processed
+independently in the order they are provided. If a pixel in the output
+file intersects pixels in multiple input files, the output pixel value will
+be derived from the final intersecting input only. If the input files do not
+overlap, combining the inputs into a :ref:`VRT file <raster.vrt>` using
+:ref:`gdalbuildvrt` will allow the output pixel to consider contributions from
+all inputs.
```

Comment:
Is this any better?


..will be derived from the final intersecting input only. If the inputs do not overlap, this can be avoided by combining the inputs into a :ref:`VRT file <raster.vrt>` (e.g., using
:ref:`gdalbuildvrt`), in which case all inputs will be used to calculate the value of output pixels.

Reply:
(The situation expressed in code)

```
from osgeo import gdal

gdal.UseExceptions()

drv = gdal.GetDriverByName("GTiff")

fname1 = '/tmp/src1.tif'
fname2 = '/tmp/src2.tif'

with drv.Create(fname1, 2, 1, 1, gdal.GDT_Float32) as ds:
    ds.GetRasterBand(1).Fill(10)
    ds.SetGeoTransform((0, 1, 0, 2, 0, -1))

with drv.Create(fname2, 2, 1, 1, gdal.GDT_Float32) as ds:
    ds.GetRasterBand(1).Fill(5)
    ds.SetGeoTransform((0, 1, 0, 1, 0, -1))

gdal.BuildVRT('/tmp/both.vrt', [fname1, fname2])

# each output pixel is 50% covered by a pixel from src1 and 50% covered by a pixel from src2

out1 = gdal.Warp('/tmp/out.tif', [fname1, fname2],
         xRes=1, yRes=1, outputBounds=(0, 0.5, 2, 1.5), resampleAlg='average')
print(out1.ReadAsArray()) # [[5. 5]] ; calculated using values from src2 only

out2 = gdal.Warp('/tmp/out_vrt.tif', '/tmp/both.vrt',
          xRes=1, yRes=1, outputBounds=(0, 0.5, 2, 1.5), resampleAlg='average')
print(out2.ReadAsArray()) # [[7.5, 7.5]] ; calculated using values from both src1 and src2
```

Reply:
oh, this is very subtle. So this is for cases on the boundary of adjacent tiles, when the output doesn't align on exact pixel boundaries of the input? Perhaps that could be mentionned because otherwise there should be little difference for the buildvrt vs multiple input sources for non-intersecting sources.

One other less subtle difference is that gdalwarp with multiple input datasets that are intersecting and have a alpha channel with values not 0 or 255 will do true alpha blending, which gdalbuildvrt isn't capable of.

Reply:
> So this is for cases on the boundary of adjacent tiles, when the output doesn't align on exact pixel boundaries of the input?

I think it's a little more broad than that, occurring even if the output pixel aligns with pixel boundaries of the input. It's any case where more than one input file would contribute to an output pixel. I was looking at "average" resampling because it's easy to understand, but you could equally imagine cubic or bilinear sampling failing to take into account pixels from an adjacent input file.

Reply:
right. So perhaps mention edge effects on source rasters ?  When I read the current text, I was focused on the "interior" of the tiles, and couldn't make sense of why buildvrt would differ from multiple inputs

Reply:
```
When multiple inputs are provided to :program:`gdalwarp`, they are processed
independently in the order they are listed. This may introduce edge effects
near the boundaries of the input files, because output pixel values will be
derived from the final input only. To avoid this, non-overlapping input
files may first be combined into a :ref:`VRT file <raster.vrt>` (e.g., using 
:ref:`gdalbuildvrt`). This will allow :program:`gdalwarp` to use pixels
from all inputs when calculating output pixel values.
```

Reply:
Agreed! Thanks for the discussion.